### PR TITLE
Partition org domain

### PR DESCRIPTION
**Pull request by snimavat**
https://github.com/9ci/domain9/issues/2810#issuecomment-2501904044

Depends on https://github.com/9ci/nine-db/pull/593

### CODE REVIEW

File: rally-domain/src/main/groovy/yakworks/rally/orgs/repo/AbstractOrgRepo.groovy [hunk 1]
```diff
@@ -98,6 +98,7 @@ abstract class AbstractOrgRepo extends LongIdGormRepo<Org> {
             // do contact, support keyContact for legacy and Customers
             def contactData = data.contact ?: data.keyContact
             if (contactData) createOrUpdatePrimaryContact(org, contactData as Map)
+            createOrUpdatePartitionOrg(org, args.bindAction)
```
Comment (basejump) [inline rally-domain/src/main/groovy/yakworks/rally/orgs/repo/AbstractOrgRepo.groovy:101]: see comment on why we do location and contact here. PartitionOrg does not belong here.

File: rally-domain/src/main/groovy/yakworks/rally/orgs/repo/AbstractOrgRepo.groovy [hunk 2]
```diff
@@ -161,6 +162,16 @@ abstract class AbstractOrgRepo extends LongIdGormRepo<Org> {
         OrgSource.query(orgId: org.getId()).deleteAll()
     }
 
+    protected void createOrUpdatePartitionOrg(Org org, BindAction bindAction) {
+        if(org.isOrgType(orgProps.partition.type)) {
+            if (bindAction == BindAction.Create) {
```
Comment (basejump) [inline rally-domain/src/main/groovy/yakworks/rally/orgs/repo/AbstractOrgRepo.groovy:167]: why only on bind?

File: rally-domain/src/main/groovy/yakworks/rally/orgs/model/PartitionOrg.groovy [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import gorm.tools.model.NameNum
+import gorm.tools.repository.model.RepoEntity
+import grails.compiler.GrailsCompileStatic
+import grails.persistence.Entity
+import yakworks.security.audit.AuditStamp
+
+@Entity
+@AuditStamp
+@GrailsCompileStatic
+class PartitionOrg implements NameNum, RepoEntity<PartitionOrg>, Serializable {
```
Comment (basejump) [inline rally-domain/src/main/groovy/yakworks/rally/orgs/model/PartitionOrg.groovy:16]: We have Client and Company too, and now PartitionOrg. We do them all different. Lets bring some consistency to it. Do we need Company? @jdabal that used at CED?

Comment (basejump) [reply rally-domain/src/main/groovy/yakworks/rally/orgs/model/PartitionOrg.groovy:16]: do we need a repo to verify here? Whats the strategy going to be for rest on this? none? Will UI be editing the PartitionOrg or only the Org which will update this?

File: rally-domain/src/main/groovy/yakworks/rally/orgs/repo/AbstractOrgRepo.groovy (hunk 2 above)
Comment (basejump) [reply rally-domain/src/main/groovy/yakworks/rally/orgs/repo/AbstractOrgRepo.groovy:167]: what happens on delete?

**Review by basejump — CHANGES_REQUESTED** [review]

File: rally-domain/src/main/groovy/yakworks/rally/orgs/model/PartitionOrg.groovy (hunk 3 above)
Comment (jdabal) [reply rally-domain/src/main/groovy/yakworks/rally/orgs/model/PartitionOrg.groovy:16]: CED is using Company (main and badDebt) and PartitionOrg (credit iffices). 
No one is using Client, even if there is one company we use company. 
So I would keep Company and PartitionOrg

File: rally-domain/src/main/groovy/yakworks/rally/orgs/repo/AbstractOrgRepo.groovy (hunk 1 above)
Comment (snimavat) [reply rally-domain/src/main/groovy/yakworks/rally/orgs/repo/AbstractOrgRepo.groovy:101]: @basejump if we do in afterPersist - thn we cant check, if num/name are dirty/updated so we can keep in sync

**Pull request merged**